### PR TITLE
testutil/promrated: add networks flag

### PR DESCRIPTION
**Pull request by LukeHackett12**
Specify networks to monitor with promrated

category: feature
ticket: #2571

### CODE REVIEW

File: testutil/promrated/promrated.go [hunk 1]
```diff
@@ -70,39 +71,37 @@ func reportMetrics(ctx context.Context, config Config) {
...
-
-		clusterLabels := prometheus.Labels{
-			"pubkey_full":     validator.PubKey,
-			"cluster_name":    validator.ClusterName,
-			"cluster_hash":    validator.ClusterHash,
-			"cluster_network": validator.ClusterNetwork,
+		if contains(config.Networks, validator.ClusterNetwork) {
+			stats, err := getValidatorStatistics(ctx, config.RatedEndpoint, config.RatedAuth, validator)
+			if err != nil {
+				log.Error(ctx, "Getting validator statistics", err, z.Str("validator", validator.PubKey))
```
Comment (xenowits) [inline testutil/promrated/promrated.go:84]:
```suggestion
				log.Error(ctx, "Getting validator statistics", err, z.Str("pubkey", validator.PubKey))
```

File: testutil/promrated/promrated.go [hunk 2]
```diff
@@ -119,3 +118,15 @@ func reportMetrics(ctx context.Context, config Config) {
 		networkEffectiveness.With(networkLabels).Set(stats.ValidatorEffectiveness)
 	}
 }
+
+func contains(arr []string, s string) bool {
```
Comment (xenowits) [inline testutil/promrated/promrated.go:122]: add godoc. Or, rather can remove this helper as it is only used once

**Review by xenowits — APPROVED** [review]
LGTM!

File: testutil/promrated/promrated.go (hunk 2 above)
Comment (LukeHackett12) [reply testutil/promrated/promrated.go:123]: I opted for godoc as I think it's still cleaner to have this extracted 👍🏼

**Pull request closed without merging**